diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index b7a1201..d3f0452 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -2,5 +2,5 @@
 ADD_SUBDIRECTORY(InterfaceOpenMVG)
 ADD_SUBDIRECTORY(DensifyPointCloud)
 ADD_SUBDIRECTORY(ReconstructMesh)
-ADD_SUBDIRECTORY(RefineMesh)
+# ADD_SUBDIRECTORY(RefineMesh)
 ADD_SUBDIRECTORY(TextureMesh)
diff --git a/apps/InterfaceOpenMVG/CMakeLists.txt b/apps/InterfaceOpenMVG/CMakeLists.txt
index f5683d1..3b1c455 100644
--- a/apps/InterfaceOpenMVG/CMakeLists.txt
+++ b/apps/InterfaceOpenMVG/CMakeLists.txt
@@ -1,7 +1,4 @@
-cxx_executable_with_flags_no_pch(InterfaceOpenMVG "Apps" "${cxx_default}" "MVS" InterfaceOpenMVG.cpp)
-if(OpenMVS_USE_BREAKPAD AND BREAKPAD_FOUND)
-	target_link_libraries(InterfaceOpenMVG ${BREAKPAD_LIBS})
-endif()
+add_definitions(-D_FILE_UTIL_H_) # HACK!
 
 FIND_PACKAGE(OpenMVG QUIET)
 IF (OPENMVG_FOUND)
@@ -10,6 +7,7 @@ IF (OPENMVG_FOUND)
 	if(OpenMVS_USE_BREAKPAD AND BREAKPAD_FOUND)
 		target_link_libraries(InterfaceOpenMVG2 ${BREAKPAD_LIBS})
 	endif()
+	target_link_libraries(InterfaceOpenMVG2 ${OpenCV_LIBS})
 ELSE()
 	MESSAGE("OPENMVG_NOT FOUND : OpenMVG importer will not be build")
 ENDIF()
diff --git a/libs/Common/Common.cpp b/libs/Common/Common.cpp
index 052951d..99f33ae 100644
--- a/libs/Common/Common.cpp
+++ b/libs/Common/Common.cpp
@@ -34,3 +34,10 @@ namespace boost {
 } // namespace boost
 #endif
 #endif
+
+// This is a super-horrible hack
+#include <opencv2/opencv.hpp>
+namespace cv {
+_OutputArray::~_OutputArray() { }
+_InputArray::~_InputArray() { }
+}
